roles/splunk_common/tasks/get_facts.yml

```diff
@@ -70,13 +70,16 @@
 - name: "Set current version fact"
   set_fact:
     splunk_current_version: "{{ manifests.files[0].path | regex_search(regexp, '\\1') if (manifests.matched == 1) else '0' }}"
+    splunk_current_build_hash: "{{manifests.files[0].path | regex_search(regexp, '\\3') if (manifests.matched == 1) else '0'}}"
+    splunk_target_build_hash: "{{ splunk.build_location | regex_search(regexp, '\\3') | default('0') }}"
   vars:
-    regexp: 'splunk\D*?-((\d+)\.(\d+)\.(\d+))'
+    regexp: 'splunk\D*?-(\d+\.\d+\.\d+(\.\d+)?)-(.*)-.*-.*.[tgz|msi|rpm|deb]'
```

Comment:
stops catching unnecessary matches and also allows for .patch versions
match 1 is version, match 3 is hash

Reply:
You may want to see if you can make these matches lazy.  It would probably be okay this way, but .*? should be more efficient